### ISSUE TITLE
Cannot resolve module 'leveldown' and 'leveldown/package' during 'npm run compile'

### ISSUE THREAD

**Issue by OutsourcedGuru** [KeitIG/museeks#316]
Being an Electron-noob this might not be an issue but I'll create one in case it is a missing dependency.  I found the installation instructions a little cryptic in that this is my first Electron attempt so the process is still foreign to me.  Poking around your README.md and that from [Electron's application distribution instructions](https://electron.atom.io/docs/tutorial/application-distribution/), I did the following workflow:

* Previously installed npm install -g electron
* Obviously, I have npm installed globally as well, v3.10.10
* Windows 7 Professional 64-bit

1. `mkdir d:\sites\mplayer\mplayer-win32-x64`
2. Extracted their **electron-v1.4.15-win32-x64.zip** from [their build repository](https://github.com/electron/electron/releases) into it
3. `mkdir d:\sites\mplayer\mplayer\win32-x64\resources\app`
4. `cd \sites\mplayer\mplayer\mplayer-win32-x64\resources\app`
5. Extracted your **https://github.com/KeitIG/archives/museeks-master.zip** into this new folder
6. Slightly edited the **package.json** to change the name/version
7. `npm install`
8. `npm run compile`

At this point during the **webpack** step, I received the following two warnings which may or may not affect the outcome:

```
WARNING in ./~/levelup/lib/util.js
Module not found: Error: Cannot resolve module 'leveldown/package' in D:\sites\mplayer\mplayer-win32-x64\resources\app\node_modules\levelup\lib
 @ ./~/levelup/lib/util.js 36:23-51
WARNING in ./~/levelup/lib/util.js
Module not found: Error: Cannot resolve module 'leveldown' in D:\sites\mplayer\mplayer-win32-x64\resources\app\node_modules\levelup\lib
 @ ./~/levelup/lib/util.js 52:23-43
```

9. cd \sites\mplayer\mplayer\mplayer-win32-x64
10. electron

I mention the workflow because it may be lost on other noobs like myself that it's necessary to run electron.exe from a few levels higher than the **app** folder.

Next, reading [their 'Generating asar archive' page](https://electron.atom.io/docs/tutorial/application-packaging/), it looks like it's possible to create a simple app.asar image and put it at the same level as d:\sites\mplayer\mplayer-win32-x64\resources in my case.  This doesn't seem to be what you're trying for in the webpack step, I'm guessing.

TL:DR
   is the main package.json missing a dependency for **leveldown**?

**Comment by YurySolovyov** [comment]
Don't worry about it, this dependency is optional.

**Comment by OutsourcedGuru** [comment]
Cool.  (Nice to know.)

Was just reading [levelUP page here](https://www.npmjs.com/package/levelup)...

```
Relationship to LevelDOWN

LevelUP is designed to be backed by LevelDOWN which provides a pure C++ binding to LevelDB and can be used as a stand-alone package if required.

As of version 0.9, LevelUP no longer requires LevelDOWN as a dependency so you must npm install leveldown when you install LevelUP.

LevelDOWN is now optional because LevelUP can be used with alternative backends, such as level.js in the browser or MemDOWN for a pure in-memory store.

LevelUP will look for LevelDOWN and throw an error if it can't find it in its Node require() path. It will also tell you if the installed version of LevelDOWN is incompatible.

The level package is available as an alternative installation mechanism. Install it instead to automatically get both LevelUP & LevelDOWN. It exposes LevelUP on its export (i.e. you can var leveldb = require('level')).
```

Personally, I find it annoying that it's complaining so much and they know it.
